backend/routes/event.js

```diff
@@ -1,95 +1,90 @@
+var router = express.Router();
 
+const auth = require("../util/auth");
 const Event = require("../models/event.model");
 
-router.get("/", (req, res, next) => {
-  Event.find({})
-    .sort({ createdAt: -1 })
-    .limit(10)
+// Creates a event
+router.post("/", auth, function (req, res, next) {
+  const event = new Event({
+    title: req.body.title,
+    community: req.body.community,
+    description: req.body.description,
+    pdfPath: req.body.pdfPath || null,
+    eventLink: req.body.eventLink || null,
+    banner: req.body.banner || null
+  });
+
+  event
+    .save()
     .then((result) => {
-      res.status(200).json({
+      res.status(201).json({
         result: result,
         message: "OK",
       });
+      console.log(result);
     })
     .catch((err) => {
       res.status(500).json({
         message: "Something went wrong",
       });
+      console.log(err);
     });
 });
 
-router.get("/all", (req, res, next) => {
+// Returns 5 the Event (sorted : latest first)
+router.get("/", (req, res, next) => {
   Event.find({})
     .sort({ createdAt: -1 })
+    .limit(5)
     .then((result) => {
       res.status(200).json({
         result: result,
-        message: "OK",
+        message: 'OK'
       });
     })
     .catch((err) => {
+      console.log(err);
       res.status(500).json({
         message: "Something went wrong",
       });
     });
 });
 
-router.post("/", auth, (req, res, next) => {
-  const event = new Event({
-    title: req.body.title,
-    description: req.body.description,
-    community: req.body.community,
-    banner: req.body.banner,
-  });
-  event
-    .save()
+// Returns all the events
+router.get("/all", (req, res, next) => {
+  Event.find({})
+    .sort({ createdAt: -1 })
     .then((result) => {
-			res.status(201).json({
+      res.status(200).json({
         result: result,
         message: 'OK'
       });
-      console.log(result);
     })
     .catch((err) => {
-      res.status(500).json({
-				message: 'Something went wrong'
-			});
       console.log(err);
+      res.status(500).json({
+        message: "Something went wrong",
+      });
     });
 });
 
-router.put('/:id', auth, (req, res, next) => {
-	const {title, description, community, banner} = req.body;
-	Event.findOneAndUpdate({_id: req.params.id}, {title, description, community, banner},)
-	.then((result) => {
-		res.status(200).json({
-			result,
-			message: 'OK'
-		})
-	})
-	.catch(err => {
-		res.status(500).json({
-			message: 'Something went Wrong'
-		})
-		console.log(err);
-	})
-})
-
+// Deletes a Event
 router.delete("/:id", auth, (req, res, next) => {
   Event.findOneAndRemove({ _id: req.params.id })
     .then(() => {
       res.status(200).json({
-        message: "OK",
+        message: 'OK'
       });
     })
     .catch((err) => {
+      console.log(err);
       res.status(500).json({
         message: "Something went wrong",
       });
     });
 });
 
+// Should we allow updation to events, requires disussion
```

Comment:
Why not? We should be adding a put route here.

Reply:
We'll have to maintain a history of every iteration!

Reply:
I don't think it is necessary at this point.. The frontend will evaluate `eventDate` after fetching and display it in the relevant table. So maintaining a record will turn out to be an overhead instead of providing functionality.

Reply:
But having the history is very important for some security or so purpose! If required I'm putting this as a feature update for the future.